cmd/clusterctl/internal/util/image.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// ImageSplit contains all the information resulting from a split of an image name into its composing parts.
+type ImageSplit struct {
+	// Repository where the image is hosted
+	Repository string
+
+	// Name of the image
+	Name string
+
+	// Tag of the image
+	Tag string
+}
+
+// NewImageSplit returns a ImageSplit starting from an image name.
+func NewImageSplit(image string) (*ImageSplit, error) {
+	ret := &ImageSplit{}
+
+	imageSplit := strings.Split(image, ":")
```

Comment:
The repository URL could have a ':' in it, in the case of `my.docker.registry:5000/my/image:latest`

Reply:
Right. Either need to get the last index, or use `github.com/docker/distribution/reference` (also coming in #2559 as it currently stands). My preference would be to use the distribution package.

Reply:
fixed